apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,64 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
 
+
+function movePiece(startStack,endStack) {  
+
+  var poppedPiece = stacks[startStack].pop();
+  stacks[endStack].push(poppedPiece);
 }
 
-function isLegal() {
-  // Your code here
 
+
+
+function isLegal(startStack, endStack) {
+  var start = stacks[startStack].length;
+  var end = stacks[endStack].length;
+
+
+
+  if (stacks[endStack].length === 0 || stacks[startStack][start - 1] < stacks[endStack][end - 1]) {
```

Comment:
Nice work!!!!

![dance](https://media.giphy.com/media/l46CdUTy21h8wwrAs/giphy.gif)

You can also pass in `if(end === 0 || ...` since you already stored what you're looking for in a variable.